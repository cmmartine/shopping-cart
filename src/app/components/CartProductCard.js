import { useDispatch } from "react-redux";
import { increment, decrement } from "../../features/productSlice";

function CartProductCard(props) {

  const {prodName, prodImg, prodCost, numInCart, id} = props.product;

  const dispatch = useDispatch();

  return(
    <div className='cart-card-inner'>

      <img src={prodImg} alt={prodName} />

      <div className='prod-info-container'>
        <h3 className='prod-name'>{prodName}</h3>

        <p className='prod-cost'>{prodCost}</p>
      </div>
      
      <div className='count-container'>
        <div>Quantity: {numInCart}</div>
        <div className='count-buttons-container'>
          <button className='cart-amount-button' aria-label='decrement' onClick={() => dispatch(decrement(id))}>-</button>
          <button className='cart-amount-button' aria-label='increment' onClick={() => dispatch(increment(id))}>+</button>
        </div>
      </div>

    </div>
  );
}

export default CartProductCard;