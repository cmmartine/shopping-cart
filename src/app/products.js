export const allProducts = [
  {
    prodName: 'Shoes',
    prodImg: 'Some image',
    prodDesc: 'Very good',
    prodCost: '$49.99',
    numInCart: 0,
    id: 1,
  },
  {
    prodName: 'Shirt',
    prodImg: 'Some image',
    prodDesc: 'Quality fabric',
    prodCost: '$19.99',
    numInCart: 0,
    id: 2,
  }
];