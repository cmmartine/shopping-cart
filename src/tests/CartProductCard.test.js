import renderer from 'react-test-renderer';
import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import CartProductCard from '../app/components/CartProductCard';
import { renderWithProviders } from './test_utils/redux-store-test-util';
import { store } from '../app/store';
import { Provider } from 'react-redux';

describe('Cart Product Card', () => {
  const product = {
    prodName: 'Shoes',
    prodImg: 'Some image',
    prodDesc: 'Very good',
    prodCost: '$49.99',
    numInCart: 0,
    id: 1,
  }

  const productInCart = {
    prodName: 'Shoes',
    prodImg: 'Some image',
    prodDesc: 'Very good',
    prodCost: '$49.99',
    numInCart: 3,
    id: 1,
  }

  it('renders correctly', () => {

    const view = renderer
      .create(
        <Provider store={store}>
        <CartProductCard product={product}/>
      </Provider>
      )
      .toJSON();
    
      expect(view).toMatchSnapshot();
  });

  it('renders the card passed to it', () => {

    renderWithProviders(<CartProductCard product={product} />);

    expect(screen.getByRole('heading').textContent).toMatch('Shoes');
  });

  it('renders correct quantity when numInCart is 0', () => {

    renderWithProviders(<CartProductCard product={product} />);

    expect(screen.getByText('Quantity: 0')).toBeInTheDocument();
  });

  it('renders correct quantity when numInCart is 3', () => {

    renderWithProviders(<CartProductCard product={productInCart} />);

    expect(screen.getByText('Quantity: 3')).toBeInTheDocument();
  });

});