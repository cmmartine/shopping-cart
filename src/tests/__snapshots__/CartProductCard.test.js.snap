// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Product Card renders correctly 1`] = `
<div
  className="cart-card-inner"
>
  <img
    alt="Shoes"
    src="Some image"
  />
  <div
    className="prod-info-container"
  >
    <h3
      className="prod-name"
    >
      Shoes
    </h3>
    <p
      className="prod-cost"
    >
      $49.99
    </p>
  </div>
  <div
    className="count-container"
  >
    <div
      className="cart-quantity"
    >
      Quantity: 
      0
    </div>
    <div
      className="count-buttons-container"
    >
      <button
        aria-label="decrement"
        className="cart-amount-button"
        onClick={[Function]}
      >
        -
      </button>
      <button
        aria-label="increment"
        className="cart-amount-button"
        onClick={[Function]}
      >
        +
      </button>
    </div>
  </div>
</div>
`;
